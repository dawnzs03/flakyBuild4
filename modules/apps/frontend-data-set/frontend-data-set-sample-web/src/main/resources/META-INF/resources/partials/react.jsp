<%--
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */
--%>

<%@ include file="/init.jsp" %>

<%
ReactFDSDisplayContext reactFDSDisplayContext = new ReactFDSDisplayContext(request);
%>

<react:component
	module="{SampleReactFrontendDataSet} from frontend-data-set-sample-web"
	props='<%=
		HashMapBuilder.<String, Object>put(
			"apiURL", reactFDSDisplayContext.getAPIURL()
		).put(
			"id", FDSSampleFDSNames.REACT
		).put(
			"style", "fluid"
		).put(
			"views", reactFDSDisplayContext.getViews()
		).build()
	%>'
/>