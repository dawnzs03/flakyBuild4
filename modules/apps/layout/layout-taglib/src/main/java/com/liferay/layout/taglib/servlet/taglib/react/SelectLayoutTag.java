/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.layout.taglib.servlet.taglib.react;

import com.liferay.layout.taglib.internal.servlet.ServletContextUtil;
import com.liferay.layout.taglib.internal.util.LayoutUtil;
import com.liferay.portal.kernel.json.JSONArray;
import com.liferay.portal.kernel.json.JSONFactoryUtil;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Group;
import com.liferay.portal.kernel.model.LayoutConstants;
import com.liferay.portal.kernel.service.LayoutServiceUtil;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.GetterUtil;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.ParamUtil;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.portal.util.PropsValues;
import com.liferay.taglib.util.IncludeTag;

import java.util.Map;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Eudaldo Alonso
 * @author Marko Cikos
 */
public class SelectLayoutTag extends IncludeTag {

	public String getItemSelectorReturnType() {
		return _itemSelectorReturnType;
	}

	public String getItemSelectorSaveEvent() {
		return _itemSelectorSaveEvent;
	}

	public String getNamespace() {
		return _namespace;
	}

	public boolean isCheckDisplayPage() {
		return _checkDisplayPage;
	}

	public boolean isEnableCurrentPage() {
		return _enableCurrentPage;
	}

	public boolean isMultiSelection() {
		return _multiSelection;
	}

	public boolean isPrivateLayout() {
		return _privateLayout;
	}

	public void setCheckDisplayPage(boolean checkDisplayPage) {
		_checkDisplayPage = checkDisplayPage;
	}

	public void setEnableCurrentPage(boolean enableCurrentPage) {
		_enableCurrentPage = enableCurrentPage;
	}

	public void setItemSelectorReturnType(String itemSelectorReturnType) {
		_itemSelectorReturnType = itemSelectorReturnType;
	}

	public void setItemSelectorSaveEvent(String itemSelectorSaveEvent) {
		_itemSelectorSaveEvent = itemSelectorSaveEvent;
	}

	public void setMultiSelection(boolean multiSelection) {
		_multiSelection = multiSelection;
	}

	public void setNamespace(String namespace) {
		_namespace = namespace;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	public void setPrivateLayout(boolean privateLayout) {
		_privateLayout = privateLayout;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_checkDisplayPage = false;
		_enableCurrentPage = false;
		_itemSelectorReturnType = null;
		_itemSelectorSaveEvent = null;
		_multiSelection = false;
		_namespace = null;
		_privateLayout = false;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			httpServletRequest.setAttribute(
				"liferay-layout:select-layout:data", _getData());
		}
		catch (Exception exception) {
			_log.error(exception);
		}
	}

	private Map<String, Object> _getData() throws Exception {
		HttpServletRequest httpServletRequest = getRequest();

		ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		String[] selectedLayoutIds = ParamUtil.getStringValues(
			httpServletRequest, "layoutUuid");

		return HashMapBuilder.<String, Object>put(
			"checkDisplayPage", _checkDisplayPage
		).put(
			"config",
			HashMapBuilder.<String, Object>put(
				"loadMoreItemsURL",
				themeDisplay.getPathMain() + "/portal/get_layouts"
			).put(
				"maxPageSize",
				GetterUtil.getInteger(
					PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN)
			).build()
		).put(
			"groupId", themeDisplay.getScopeGroupId()
		).put(
			"itemSelectorReturnType", _itemSelectorReturnType
		).put(
			"itemSelectorSaveEvent", _itemSelectorSaveEvent
		).put(
			"multiSelection", _multiSelection
		).put(
			"namespace", _namespace
		).put(
			"nodes", _getLayoutsJSONArray(selectedLayoutIds, themeDisplay)
		).put(
			"privateLayout", _privateLayout
		).put(
			"selectedLayoutIds", selectedLayoutIds
		).build();
	}

	private JSONArray _getLayoutsJSONArray(
			String[] selectedLayoutIds, ThemeDisplay themeDisplay)
		throws Exception {

		Group group = themeDisplay.getScopeGroup();

		if ((_privateLayout && !group.hasPrivateLayouts()) ||
			(!_privateLayout && !group.hasPublicLayouts())) {

			return JSONFactoryUtil.createJSONArray();
		}

		return JSONUtil.put(
			JSONUtil.put(
				"children",
				LayoutUtil.getLayoutsJSONArray(
					_checkDisplayPage, _enableCurrentPage,
					themeDisplay.getScopeGroupId(), getRequest(),
					_itemSelectorReturnType, _privateLayout, 0,
					selectedLayoutIds, 0,
					GetterUtil.getInteger(
						PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN))
			).put(
				"disabled", true
			).put(
				"expanded", true
			).put(
				"hasChildren", true
			).put(
				"icon", "home"
			).put(
				"id", "0"
			).put(
				"name", themeDisplay.getScopeGroupName()
			).put(
				"paginated",
				() -> {
					int layoutsCount = LayoutServiceUtil.getLayoutsCount(
						themeDisplay.getScopeGroupId(), _privateLayout,
						LayoutConstants.DEFAULT_PARENT_LAYOUT_ID);

					if (layoutsCount >
							PropsValues.LAYOUT_MANAGE_PAGES_INITIAL_CHILDREN) {

						return true;
					}

					return false;
				}
			));
	}

	private static final String _PAGE = "/select_layout/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(
		SelectLayoutTag.class);

	private boolean _checkDisplayPage;
	private boolean _enableCurrentPage;
	private String _itemSelectorReturnType;
	private String _itemSelectorSaveEvent;
	private boolean _multiSelection;
	private String _namespace;
	private boolean _privateLayout;

}