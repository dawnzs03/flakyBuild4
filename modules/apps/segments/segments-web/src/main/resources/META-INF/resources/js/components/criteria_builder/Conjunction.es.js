/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import ClayButton from '@clayui/button';
import ClayDropdown from '@clayui/drop-down';
import ClayIcon from '@clayui/icon';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, {useEffect, useState} from 'react';

import {SUPPORTED_CONJUNCTIONS} from '../../utils/constants';

function Conjunction({className, conjunctionName, editing, onSelect}) {
	const [active, setActive] = useState(false);

	const classnames = classNames(
		{
			'conjunction-button': editing,
			'conjunction-label': !editing,
		},
		className
	);

	const [activeLabel, setActiveLabel] = useState(null);
	useEffect(() => {
		const selectedConjunction = SUPPORTED_CONJUNCTIONS.find(
			(conjunction) =>
				conjunction.name.toLowerCase() === conjunctionName.toLowerCase()
		);

		setActiveLabel(selectedConjunction.label);
	}, [conjunctionName]);

	function _handleItemClick(conjunctionName) {
		setActive(false);

		onSelect(conjunctionName);
	}

	return editing ? (
		<ClayDropdown
			active={active}
			className={classnames}
			onActiveChange={setActive}
			trigger={
				<ClayButton
					className="text-capitalize"
					displayType="secondary"
					small
				>
					{activeLabel}

					<ClayIcon className="ml-2" symbol="caret-bottom" />
				</ClayButton>
			}
		>
			<ClayDropdown.ItemList>
				{SUPPORTED_CONJUNCTIONS.map((conjunction) => {
					return (
						<ClayDropdown.Item
							className="text-capitalize"
							key={conjunction.name}
							onClick={() => _handleItemClick(conjunction.name)}
						>
							{conjunction.label}
						</ClayDropdown.Item>
					);
				})}
			</ClayDropdown.ItemList>
		</ClayDropdown>
	) : (
		<div className={classnames}>{activeLabel}</div>
	);
}

Conjunction.propTypes = {
	className: PropTypes.string,
	conjunctionName: PropTypes.string.isRequired,
	editing: PropTypes.bool.isRequired,
	onSelect: PropTypes.func.isRequired,
};

export default Conjunction;
