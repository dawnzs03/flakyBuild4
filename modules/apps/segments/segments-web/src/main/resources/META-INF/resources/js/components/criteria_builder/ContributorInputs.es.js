/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import PropTypes from 'prop-types';
import React from 'react';

import {initialContributorShape} from '../../utils/types.es';

function ContributorInputs({contributors}) {
	return contributors
		.filter((criteria) => criteria.query)
		.map((criteria, i) => {
			return (
				<React.Fragment key={i}>
					<input
						className="field form-control"
						data-testid={criteria.inputId}
						id={criteria.inputId}
						name={criteria.inputId}
						readOnly
						type="hidden"
						value={criteria.query}
					/>

					<input
						id={criteria.conjunctionInputId}
						name={criteria.conjunctionInputId}
						readOnly
						type="hidden"
						value={criteria.conjunctionId}
					/>
				</React.Fragment>
			);
		});
}

ContributorInputs.propTypes = {
	contributors: PropTypes.arrayOf(initialContributorShape),
};

export default ContributorInputs;
