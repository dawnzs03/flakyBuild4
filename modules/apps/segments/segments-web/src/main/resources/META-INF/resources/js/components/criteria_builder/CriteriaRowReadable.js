/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {PropTypes} from 'prop-types';
import React from 'react';

import {PROPERTY_TYPES} from '../../utils/constants';
import {unescapeSingleQuotes} from '../../utils/odata';
import {dateToInternationalHuman} from '../../utils/utils';

export default function CriteriaRowReadable({
	criterion = {},
	selectedOperator,
	selectedProperty,
}) {
	const _renderCriteriaString = ({
		operatorLabel,
		propertyLabel,
		type,
		value,
	}) => {
		let parsedValue = null;

		if (type === PROPERTY_TYPES.DATE) {
			parsedValue = dateToInternationalHuman(
				value.replaceAll
					? value.replaceAll('-', '/')
					: value.replace(/-/g, '/')
			);
		}
		else if (type === PROPERTY_TYPES.DATE_TIME) {
			parsedValue = dateToInternationalHuman(value);
		}
		else {
			parsedValue = value;
		}

		return (
			<span>
				{propertyLabel && (
					<b className="c-mr-1 font-weight-bold text-dark">
						{propertyLabel}
					</b>
				)}

				{operatorLabel && (
					<span className="c-mr-1 font-weight-bold operator">
						{operatorLabel}
					</span>
				)}

				<b>{unescapeSingleQuotes(parsedValue)}</b>
			</span>
		);
	};

	const value = criterion ? criterion.value : '';
	const operatorLabel = selectedOperator ? selectedOperator.label : '';
	const propertyLabel = selectedProperty ? selectedProperty.label : '';

	return (
		<span className="c-ml-2 criterion-string">
			{_renderCriteriaString({
				operatorLabel,
				propertyLabel,
				type: selectedProperty.type,
				value: criterion.displayValue || value,
			})}
		</span>
	);
}

CriteriaRowReadable.propTypes = {
	criterion: PropTypes.object.isRequired,
	selectedOperator: PropTypes.object,
	selectedProperty: PropTypes.object.isRequired,
};
