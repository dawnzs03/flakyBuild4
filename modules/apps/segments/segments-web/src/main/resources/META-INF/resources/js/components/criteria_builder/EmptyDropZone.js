/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import {DropTarget as dropTarget} from 'react-dnd';

import {DragTypes} from '../../utils/drag-types';
import EmptyPlaceholder from './EmptyPlaceholder.es';

/**
 * Prevents items from being dropped from other contributors.
 * This method must be called `canDrop`.
 * @param {Object} props Component's current props.
 * @param {DropTargetMonitor} monitor
 * @returns {boolean} True if the target should accept the item.
 */
function canDrop(props, monitor) {
	const {propertyKey: destPropertyKey} = props;
	const {propertyKey: startPropertyKey} = monitor.getItem();

	return destPropertyKey === startPropertyKey;
}

/**
 * Implements the behavior of what will occur when an item is dropped.
 * Adds the criterion dropped.
 * This method must be called `drop`.
 * @param {Object} props Component's current props.
 * @param {DropTargetMonitor} monitor
 */
function drop(props, monitor) {
	const {criterion} = monitor.getItem();

	props.onCriterionAdd(0, criterion);
}

function EmptyDropZone({canDrop, connectDropTarget, emptyContributors, hover}) {
	const displayEmptyDropZone = canDrop || !emptyContributors;

	return connectDropTarget(
		displayEmptyDropZone ? (
			<div
				className={classNames('empty-drop-zone p-6 rounded', {
					'border-primary': canDrop || hover,
					'border-secondary-light': !canDrop,
					'empty-drop-zone--dashed':
						!emptyContributors || (canDrop && !hover),
					'empty-drop-zone--target': hover,
				})}
			/>
		) : (
			<div>
				<EmptyPlaceholder />
			</div>
		)
	);
}

EmptyDropZone.propTypes = {
	canDrop: PropTypes.bool,
	connectDropTarget: PropTypes.func,
	emptyContributors: PropTypes.bool,
	hover: PropTypes.bool,
	onCriterionAdd: PropTypes.func.isRequired,
	propertyKey: PropTypes.string.isRequired,
};

export default dropTarget(
	DragTypes.PROPERTY,
	{
		canDrop,
		drop,
	},
	(connect, monitor) => ({
		canDrop: monitor.canDrop(),
		connectDropTarget: connect.dropTarget(),
		hover: monitor.isOver(),
	})
)(EmptyDropZone);
