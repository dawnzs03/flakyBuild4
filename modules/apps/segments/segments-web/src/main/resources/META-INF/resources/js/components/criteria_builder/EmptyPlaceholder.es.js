/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import ClayEmptyState from '@clayui/empty-state';
import React from 'react';

import {sub} from '../../utils/utils';

export default function EmptyPlaceholder() {
	return (
		<div className="c-mb-0 c-p-4 empty-contributors rounded">
			<ClayEmptyState
				description={Liferay.Language.get('empty-conditions-message')}
				imgSrc={`${themeDisplay.getPathThemeImages()}/states/empty_state.gif`}
				title={sub(Liferay.Language.get('no-x-yet'), [
					Liferay.Language.get('conditions'),
				])}
			></ClayEmptyState>
		</div>
	);
}
