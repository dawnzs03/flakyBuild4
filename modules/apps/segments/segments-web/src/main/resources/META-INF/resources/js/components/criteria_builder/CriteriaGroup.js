/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import ClayIcon from '@clayui/icon';
import classNames from 'classnames';
import {PropTypes} from 'prop-types';
import React, {Fragment, useMemo} from 'react';
import {DragSource as dragSource} from 'react-dnd';

import {
	CONJUNCTIONS,
	SUPPORTED_OPERATORS,
	SUPPORTED_PROPERTY_TYPES,
} from '../../utils/constants';
import {DragTypes} from '../../utils/drag-types';
import {
	generateGroupId,
	getChildGroupIds,
	getSupportedOperatorsFromType,
	insertAtIndex,
	replaceAtIndex,
} from '../../utils/utils';
import Conjunction from './Conjunction.es';
import CriteriaRow from './CriteriaRow';
import DropZone from './DropZone';
import EmptyDropZone from './EmptyDropZone';

/**
 * Passes the required values to the drop target.
 * This method must be called `beginDrag`.
 * @param {Object} props Component's current props
 * @returns {Object} The props to be passed to the drop target.
 */
function beginDrag({criteria, index, parentGroupId}) {
	const childGroupIds = getChildGroupIds(criteria);

	return {
		childGroupIds,
		criterion: criteria,
		groupId: parentGroupId,
		index,
	};
}

/**
 * A function that decorates the passed in component with the drag source HOC.
 * This was separated out since this function needed to be called again for the
 * nested groups.
 * @param {React.Component} component The component to decorate.
 */
const withDragSource = dragSource(
	DragTypes.CRITERIA_GROUP,
	{
		beginDrag,
	},
	(connect, monitor) => ({
		connectDragPreview: connect.dragPreview(),
		connectDragSource: connect.dragSource(),
		draggingItem: monitor.getItem(),
		isDragging: monitor.isDragging(),
	})
);

function CriteriaGroup({
	connectDragPreview,
	connectDragSource,
	criteria,
	draggingItem,
	editing,
	emptyContributors,
	entityName,
	groupId,
	isDragging,
	modelLabel,
	onChange,
	onMove,
	propertyKey,
	renderEmptyValuesErrors = false,
	root = false,
	supportedProperties,
}) {
	const NestedCriteriaGroupWithDrag = useMemo(() => {
		return withDragSource(CriteriaGroup);
	}, []);

	const _handleConjunctionSelect = (conjunctionName) => {
		onChange({
			...criteria,
			conjunctionName,
		});
	};

	/**
	 * Adds a new criterion in a group at the specified index. If the criteria
	 * was previously empty and is being added to the root group, a new group
	 * will be added as well.
	 * @param {number} index The position the criterion will be inserted in.
	 * @param {object} criterion The criterion that will be added.
	 * @memberof CriteriaGroup
	 */
	const _handleCriterionAdd = (index, criterion) => {
		const {
			defaultValue = '',
			operatorName,
			propertyName,
			type,
			value,
		} = criterion;

		const criterionValue = value || defaultValue;

		const operators = getSupportedOperatorsFromType(
			SUPPORTED_OPERATORS,
			SUPPORTED_PROPERTY_TYPES,
			type
		);

		const newCriterion = {
			operatorName: operatorName ? operatorName : operators[0].name,
			propertyName,
			type,
			value: criterionValue,
		};

		if (root && !criteria) {
			onChange({
				conjunctionName: CONJUNCTIONS.AND,
				groupId: generateGroupId(),
				items: [newCriterion],
			});
		}
		else {
			onChange({
				...criteria,
				items: insertAtIndex(newCriterion, criteria.items, index),
			});
		}
	};

	const _handleCriterionChange = (index) => (newCriterion) => {
		onChange({
			...criteria,
			items: replaceAtIndex(newCriterion, criteria.items, index),
		});
	};

	const _handleCriterionDelete = (index) => {
		onChange({
			...criteria,
			items: criteria.items.filter((fItem, fIndex) => fIndex !== index),
		});
	};

	const _isCriteriaEmpty = () => {
		return criteria ? !criteria.items.length : true;
	};

	const _renderConjunction = (index) => {
		return (
			<>
				<DropZone
					dropIndex={index}
					groupId={groupId}
					onCriterionAdd={_handleCriterionAdd}
					onMove={onMove}
					propertyKey={propertyKey}
				/>

				<Conjunction
					conjunctionName={criteria.conjunctionName}
					editing={editing}
					onSelect={_handleConjunctionSelect}
				/>

				<DropZone
					before
					dropIndex={index}
					groupId={groupId}
					onCriterionAdd={_handleCriterionAdd}
					onMove={onMove}
					propertyKey={propertyKey}
				/>
			</>
		);
	};

	const _renderCriterion = (criterion, index) => {
		return (
			<div
				className={classNames('criterion position-relative', {
					'criterion-group': criterion.items,
				})}
			>
				{criterion.items ? (
					<NestedCriteriaGroupWithDrag
						criteria={criterion}
						editing={editing}
						entityName={entityName}
						groupId={criterion.groupId}
						index={index}
						modelLabel={modelLabel}
						onChange={_handleCriterionChange(index)}
						onMove={onMove}
						parentGroupId={groupId}
						propertyKey={propertyKey}
						renderEmptyValuesErrors={renderEmptyValuesErrors}
						supportedProperties={supportedProperties}
					/>
				) : (
					<CriteriaRow
						criterion={criterion}
						editing={editing}
						entityName={entityName}
						groupId={groupId}
						index={index}
						modelLabel={modelLabel}
						onAdd={_handleCriterionAdd}
						onChange={_handleCriterionChange(index)}
						onDelete={_handleCriterionDelete}
						onMove={onMove}
						propertyKey={propertyKey}
						renderEmptyValuesErrors={renderEmptyValuesErrors}
						root={root}
						supportedProperties={supportedProperties}
					/>
				)}

				<DropZone
					dropIndex={index + 1}
					groupId={groupId}
					onCriterionAdd={_handleCriterionAdd}
					onMove={onMove}
					propertyKey={propertyKey}
				/>
			</div>
		);
	};

	const singleRow = criteria && criteria.items && criteria.items.length === 1;

	const disabled =
		isDragging ||
		(draggingItem && draggingItem.propertyKey !== propertyKey);

	return connectDragPreview(
		<div
			className={classNames(
				{
					'criteria-group-root w-100': criteria,
					disabled,
				},
				`color--${propertyKey} criteria-group-item${
					root ? '-root' : ''
				}`
			)}
		>
			{_isCriteriaEmpty() ? (
				<EmptyDropZone
					emptyContributors={emptyContributors}
					onCriterionAdd={_handleCriterionAdd}
					propertyKey={propertyKey}
				/>
			) : (
				<>
					<DropZone
						before
						dropIndex={0}
						groupId={groupId}
						onCriterionAdd={_handleCriterionAdd}
						onMove={onMove}
						propertyKey={propertyKey}
					/>

					{editing &&
						singleRow &&
						!root &&
						connectDragSource(
							<div className="align-items-center d-flex drag-icon h-100 position-absolute top-0">
								<ClayIcon symbol="drag" />
							</div>
						)}

					{criteria.items &&
						criteria.items.map((criterion, index) => {
							return (
								<Fragment key={index}>
									{index !== 0 && _renderConjunction(index)}

									{_renderCriterion(criterion, index)}
								</Fragment>
							);
						})}
				</>
			)}
		</div>
	);
}

CriteriaGroup.propTypes = {
	connectDragPreview: PropTypes.func,
	connectDragSource: PropTypes.func,
	criteria: PropTypes.object,
	dragging: PropTypes.bool,
	editing: PropTypes.bool,
	emptyContributors: PropTypes.bool,
	entityName: PropTypes.string,
	groupId: PropTypes.string,
	index: PropTypes.number,
	modelLabel: PropTypes.string,
	onChange: PropTypes.func,
	onMove: PropTypes.func,
	parentGroupId: PropTypes.string,
	propertyKey: PropTypes.string.isRequired,
	renderEmptyValuesErrors: PropTypes.bool,
	root: PropTypes.bool,
	supportedProperties: PropTypes.array,
};

export default withDragSource(CriteriaGroup);
