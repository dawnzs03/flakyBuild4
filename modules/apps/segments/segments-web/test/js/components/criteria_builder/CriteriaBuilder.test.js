/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';
import {wrapInTestContext} from 'react-dnd-test-utils';

import {default as Component} from '../../../../src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaBuilder';

const CriteriaBuilder = wrapInTestContext(Component);

describe('CriteriaBuilder', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {asFragment} = render(
			<CriteriaBuilder
				editing={false}
				editingCriteria={false}
				emptyContributors={false}
				entityName="User"
				id="0"
				propertyKey="user"
				supportedProperties={[]}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
