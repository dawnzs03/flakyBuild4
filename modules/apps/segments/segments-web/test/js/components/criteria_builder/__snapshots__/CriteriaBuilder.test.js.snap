// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaBuilder renders 1`] = `
<DocumentFragment>
  <div
    class="criteria-builder-root"
  >
    <h3
      class="sheet-subtitle"
    >
      x-with-property-x
    </h3>
    <div
      class="color--user criteria-group-item-root"
    >
      <div
        class="empty-drop-zone p-6 rounded border-secondary-light empty-drop-zone--dashed"
      />
    </div>
  </div>
</DocumentFragment>
`;
